GEMB/managelayers.m

```diff
@@ -35,19 +35,23 @@
 n=length(T);
 zY2=zY;
 dzMin2=zeros(size(dz));
+dzMax2=zeros(size(dz));
 
 X=1;
 Zcum = cumsum(dz); 
+
 % check if depth is too small
-dzMin2(1)=dzMin;
-for i=2:n
-    if (Zcum(i)<=zTop+Dtol)
-        dzMin2(i)=dzMin;
-        X=i;
-    else
-        dzMin2(i)=zY2*dzMin2(i-1);
-    end
-end
+ztop=(Zcum<=zTop+Dtol);
+% define minimum layer thickness for top layers
+dzMin2(find(ztop))=dzMin;
+% define maximum layer thickness for top layers
+dzMax2(find(ztop))=dzMin*2;
+
+zYcum = cumprod(zY2*ones(size(find(~ztop))));

```

Comment:
Oh good call on using `cumprod`. That's definitely the way to go. However, I think there's a simpler way to create `dzMin2`, and we can do it in a way that eliminates the need for `find` (because `find` is much more expensive than a simple logical comparison). 

This is what I'm thinking: 
```matlab 
% A logical "mask" that indicates which cells are the top layer: 
is_top_layer = Zcum <= (zTop + Dtol); 

% Define dzMin2 array using the top-layer dzMin value for the entire column:  
dzMin2 = dzMin * ones(size(dz)); 

% Overwrite the bottom layer as the cumulative product times the stretching factor: 
dzMin2(~is_top_layer) = cumprod(zY2 * dzMin2(~is_top_layer)); 
```